src/sage/rings/polynomial/multi_polynomial.pyx

```diff
@@ -20,6 +20,9 @@ from itertools import chain, islice
 from sage.misc.misc_c import prod
 
 def is_MPolynomial(x):
+    from sage.misc.superseded import deprecation
+    deprecation(32709, "the function is_MPolynomial is deprecated; use isinstance(x, sage.structure.element.MPolynomial) instead")
```

Comment:
the deprecation message is wrong:
it should be `sage.rings.polynomial.multi_polynomial.MPolynomial`, no?